src/Psalm/Internal/Analyzer/Statements/Expression/Call/FunctionCallAnalyzer.php

```diff
@@ -235,6 +243,45 @@ public static function analyze(
             );
 
             $config->eventDispatcher->dispatchAfterEveryFunctionCallAnalysis($event);
+
+            if ($is_first_class_callable) {
```

Comment:
Why are you returning early here?

Reply:
I wasn't sure about this, but it seemed the code below was generally analyzing a call to the given function, which isn't actually happening. Some guidance on what still needs to be called from the code below would be appreciated.

Reply:
you wouldn't go that far anyway, you have another `if ($is_first_class_callable)` just below that create the closure. I just found weird to return before that only when this condition `if ($function_name instanceof PhpParser\Node\Name && $function_call_info->function_id) {` is met and not otherwise